veles/secrets/gcpoauth2client/detector.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcpoauth2client
+
+import (
+	"regexp"
+
+	"github.com/google/osv-scalibr/veles"
+)
+
+const (
+	// maxContextLen is the maximum number of bytes to group related client IDs and secrets.
+	// Although Client ID and Client Secret are not long, they can appear in files far apart from each other.
+	// Seaching blocks of 10 KiB should be sufficient to find related credentials.
+	maxContextLen = 10 * 1 << 10 // 10 KiB
+)
+
+var (
+	// clientIDRe is a regular expression that matches GCP OAuth2 client IDs.
+	// There is no official documentation on the exact format of GCP OAuth2 client IDs.
+	// But example on official docs include examples that end with .apps.googleusercontent.com:
+	// - https://developers.google.com/identity/protocols/oauth2/web-server
+	//
+	// Other references also suggest similar formats:
+	// - https://gofastmcp.com/integrations/google
+	// - https://web.archive.org/web/20250418010928/https://docs.gitguardian.com/secrets-detection/secrets-detection-engine/detectors/specifics/google_oauth2_keys
+	clientIDRe = regexp.MustCompile(`[0-9]{10,15}-[a-zA-Z0-9_-]+\.apps\.googleusercontent\.com`)
+
+	// clientSecretRe is a regular expression that matches GCP OAuth2 client secrets.
+	// There is no clear documentation on the exact format of GCP OAuth2 client secrets.
+	// But most online references suggest they start with "GOCSPX-" prefix.
+	// This is a good start as it reduces false positives.
+	// References:
+	// - https://gofastmcp.com/integrations/google
+	// - https://web.archive.org/web/20250418010928/https://docs.gitguardian.com/secrets-detection/secrets-detection-engine/detectors/specifics/google_oauth2_keys
+	clientSecretRe = regexp.MustCompile(`GOCSPX-[a-zA-Z0-9_-]{10,40}`)
+)
+
+// detector implements OAuth2 client credentials detection.
+type detector struct {
+	maxContextLen int
+}
+
+// match represents a regex match with its value and position.
+type match struct {
+	value    string
+	position int
+}
+
+// NewDetector returns a detector that matches GCP OAuth2 client credentials.
+func NewDetector() veles.Detector {
+	return &detector{
+		maxContextLen: maxContextLen,
+	}
+}
+
+// MaxSecretLen returns the maximum length of secrets this detector can find.
+func (d *detector) MaxSecretLen() uint32 {
+	return uint32(d.maxContextLen)
```

Comment:
MaxSecretLen should > maxContextLen (longest allowed distance)? how about 2 * d.maxContextLen?